### PR TITLE
feat: Change the file name convention when downloading pod logs

### LINKED ISSUE #19937: Change the file name convention when downloading pod logs
# Summary

Change the log file name from `<CONTAINER>.txt` to `<APP_NAMEPSACE>-<POD_NAME>.txt`

# Motivation

When downloading the logs of a pod within an application, the default is set to `<CONTAINER>.txt`.
But in case you have several pods with the same container name, or several pods across several apps, this becomes confusing, and local files get piled up without any distinction.

# Proposal

Change the related function (attaching my PR here)

### PR DESCRIPTION
**Pull request by itay-nvn-nv**
Fixes https://github.com/argoproj/argo-cd/issues/19937

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by reggie-k — COMMENTED** [review]
I think we shouldn't give up the container name entirely, for cases of pods with sidecars.
Can we append the container name to the ns and pod in the file name?

File: pkg/apiclient/application/forwarder_overwrite.go [hunk 1]
```diff
@@ -115,9 +115,11 @@ func init() {
 	logsForwarder := func(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w gohttp.ResponseWriter, req *gohttp.Request, recv func() (proto.Message, error), opts ...func(context.Context, gohttp.ResponseWriter, proto.Message) error) {
 		if req.URL.Query().Get("download") == "true" {
 			w.Header().Set("Content-Type", "application/octet-stream")
+			appNamespace := req.URL.Query().Get("appNamespace")
+			podName := req.URL.Query().Get("podName")
 			fileName := "log"
-			if container := req.URL.Query().Get("container"); len(container) > 0 && kube.IsValidResourceName(container) {
-				fileName = container
+			if kube.IsValidResourceName(appNamespace) && kube.IsValidResourceName(podName) {
```
Comment (nitishfy) [inline pkg/apiclient/application/forwarder_overwrite.go:121]:
```suggestion
			if kube.IsValidResourceName(appNamespace) && kube.IsValidResourceName(podName) {
```

`appNameswpace` is argocd always (until you change the namespace where argocd has been installed). Instead, the namespace should be the one that the pod is part of. Furthermore, testing the above changes will give me a filename as `argocd-my-app-dev-798dbbd986-kc2vz.log`. If my pod wasn't part of deployment, I'd want to have log file in this format - `<namespace that pod is part of>-<pod-name>`.

**Review by nitishfy — CHANGES_REQUESTED** [review]
PTAL. 

I also agree with @reggie-k comments for the same. Thanks!

**Review by gdsoumya — CHANGES_REQUESTED** [review]

**Review by gdsoumya — APPROVED** [review]
LGTM

**Pull request merged**